signingscript/src/signingscript/script.py

```diff
@@ -76,7 +76,7 @@
             copy_to_dir(path_dict["full_path"], context.config["work_dir"], target=path)
             log.info("signing %s", path)
             output_files = await sign(
-                context, os.path.join(work_dir, path), path_dict["formats"]
+                context, os.path.join(work_dir, path), path_dict["formats"], comment=path_dict.get('comment')
```

Comment:
Hm. I don't love how we're surfacing authenticode-specific args in generic functions.
Would something like this work?

```
    full_path = path_dict.pop("full_path")
    signing_formats = path_dict.pop("formats")
    ...
    output_files = await sign(context, os.path.join(work_dir, path), signing_formats, **path_dict)

async def sign(context, path, signing_formats, **kwargs):
    ...
    for fmt in signing_formats:
        ...
        output = await signing_func(context, output, fmt, **kwargs)
```

This basically passes on anything that's in `path_dict` to the `signing_func` without having to know any details. We're already checking to see if `authenticode` is in the format before adding `comment` to `path_dict` in `build_filelist_dict`, so we don't necessarily need to check for it again in `sign`. Also, this allows for adding additional kwargs to other formats later on without extending this `signing_func` line any further.

We could keep the `comment=None` in `sign_authenticode`, which would cause us to error out if we send an unexpected kwarg that way.

Reply:
I think this could cause problems in the cases where we have one file with multiple signing formats...
```
      "formats": [
        "autograph_authenticode",
        "autograph_widevine",
        "autograph_omnija"
      ],
      "paths": [
        "public/build/target.zip"
      ],
      "taskId": "cJEuj31sQk29jEASng_e5w",
      "taskType": "build"
```
(from https://tools.taskcluster.net/groups/C_rQ8yseSTun4r6QRfGDsQ/tasks/HuHr-aB9QMCR-qVE-9LGUQ/details)

@escapewindow thoughts?

Reply:
True. We could make it `authenticode_comment` and take `**kwargs` in all signing functions, only paying attention to the kwargs that each cares about. I'm not sure if that's a win or not.